File: Leetcode Challenge/April/Java/Swapping Nodes in a Linked List.java
class Solution 
{
    public ListNode swapNodes(ListNode head, int k) 
    {
        ListNode curr = head;
        ListNode pointer1 = head;
        ListNode pointer2= head;
        int count = 1;
        
        while( curr != null )
        {
            if( count < k )
            {
                pointer1 = pointer1.next;
            }
            if( count > k )
            {
                pointer2 = pointer2.next;
            }
            curr = curr.next;
            count++;
        }
        
        int temp = pointer1.val;
        pointer1.val = pointer2.val;
        pointer2.val = temp;
        
        return head;
    }
}

File: Leetcode Challenge/April/Python/Swapping Nodes in a Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        
        ptr1 = head
        ptr2 = head
        kth = None
        k-=1
        while (k):
            ptr1 = ptr1.next
            k-=1
            
        kth = ptr1
        ptr1 = ptr1.next;
        
        while (ptr1) :
            ptr1 = ptr1.next;
            ptr2 = ptr2.next;
        
        ptr2.val, kth.val = kth.val, ptr2.val
        return head
